drivers/flash/spi_nor.c

```diff
@@ -1515,7 +1662,13 @@ static int spi_nor_init(const struct device *dev)
 	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
 		struct spi_nor_data *const driver_data = dev->data;
 
+#ifdef CONFIG_SPI_NOR_SUSPEND_RESUME
+		k_mutex_init(&driver_data->ownership);
+		k_sem_init(&driver_data->write_erase_sem, 1, K_SEM_MAX_LIMIT);
+		k_sem_init(&driver_data->read_sem, 1, K_SEM_MAX_LIMIT);
+#else
```

Comment:
I do not think that you can suspend more than one erase/program operation at once.

Reply:
Some devices allow it, but you have to check the bfp in the 12th DWORD. Some allow doing a erase during a program und vise versa, but no nesting. But it might be simpler for now to limit it to one.